sanic/response.py

```diff
@@ -142,13 +141,62 @@ def cookies(self):
         return self._cookies
 
 
+class ContentRangeHandler:
+    """
+    This class is for parsing the request header
+    """
+    __slots__ = ('start', 'end', 'size', 'total', 'headers')
+
+    def __init__(self, request, stats):
+        self.start = self.size = 0
+        self.end = None
+        self.headers = dict()
+        self.total = stats.st_size
+        _range = request.headers.get('Range')
+        if _range:
+            self.start, self.end = ContentRangeHandler.parse_range(_range)
+            if self.start is not None and self.end is not None:
```

Comment:
Is there a way to simplify this logic? It seems like a lot of conditionals for this?